### PR TITLE
feat: Add mutli language support

### PR DESCRIPTION
**Pull request by rgraue**
Desc:
Add multi language support

Technical Changes:
- `Language_support.py`
    - The `LanguageSupportService` now acts as the wrapper class for `pre-commit.py` and all calls are made through language support. With the introduction of multiple languages, the `LanguageSupportService` now facilitates the higher control for all of the languages, such as the total repo config, versioning, and other repo wide execution.
- `pre-commi.py`
    - The `pre-commit` abstraction's sole responsibility is that of one given language. Its job is to find, and calculate a single pre-commit config and its respective linter configs, as well as executing any pre-commit installs for the language specific hooks.

TL/DR;
Control flow: 
  Actions -> LanguageSupportService -> PreCommitAbstraction

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,5 +1,21 @@
-exclude: ^(?:.+/)?\.idea(?P<ps_d>/).*$
```
Comment (stujfiter) [inline .pre-commit-config.yaml:1]: Was this done on purpose? Changes to this file seem to be purely organizational.  Are there any functional changes?

**Comment by stujfiter** [comment]
This is a massive PR. In my experience, massive PRs like this tend to be troublesome especially when it comes to unintentionally introducing breaking changes.  Is there any way that it can be broken up into several smaller PRs? Perhaps split the refactoring from the functional changes. Can the functional changes be introduced in stages? I would feel more comfortable if some additional BATS tests were added prior to this change to ensure that the existing functionality still works after this PR.
